e2e/app/element/base-element.ts

```diff
@@ -60,10 +60,7 @@ export default class BaseElement extends Container {
    *  return await handle.jsonValue();
    */
   async getProperty(propertyName: string): Promise<unknown> {
-    if (this.element == null) {
-      throw new Error('The element is undefined.');
-    }
-    const p = await this.element.asElement().getProperty(propertyName);
+    const p = await (await this.asElementHandle()).getProperty(propertyName);
```

Comment:
In the future, if large refactoring changes like this can be split out from the PR, that would be very helpful for giving quicker review turnaround. To review 500+ LOC all at once, I need to set aside a significant amount of time. In comparison, ~400 lines of mechanical refactoring changes alone is easy - and 100 lines of new tests is easy.